### PR TITLE
static-build: Use apply_patches.sh in qemu and kernel scripts 

### PR DESCRIPTION
**Pull request by wainersm**
Creates a common script to the qemu and qemu-virtiofs build
processes to ensure the patches are applied of same way.

Backport of github.com/kata-containers/kata-containers/pull/787,
with the three commits squashed.

Fixes: #786

Signed-off-by: Wainer dos Santos Moschetta <wainersm@redhat.com>

### CODE REVIEW

File: qemu/apply_patches.sh [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+script_dir="$(realpath $(dirname $0))"
+
+qemu_version="$(cat VERSION)"
+stable_branch=$(echo $qemu_version | \
+	awk 'BEGIN{FS=OFS="."}{print $1 "." $2 ".x"}')
+patches_dir="${script_dir}/patches/${stable_branch}"
+
+echo "Handle patches for QEMU $qemu_version (stable ${stable_branch})"
+if [ -d $patches_dir ]; then
+	for patch in $(find $patches_dir -name '*.patch'); do
```
Comment (jodh-intel) [inline qemu/apply_patches.sh:21]: Given the number of bugs we've had in the past with applying patches, I'd suggest moving this to the top-level `scripts/` directory and allow `$patches_dir` to be a CLI param to this script.

But note that this script isn't reliable as it isn't sorting the patches numerically (as you do manually below).

File: static-build/qemu-virtiofs/Dockerfile [hunk 2]
```diff
@@ -49,23 +49,17 @@ RUN cd  .. && git clone "${QEMU_VIRTIOFS_REPO}" qemu-virtiofs
...
-RUN find "$(cat patches_qemu_dir)" -name '*.patch' -type f |sort -t- -k1,1n > patches_qemu
-RUN echo "Patches to apply for qemu:"
-RUN cat patches_qemu
-RUN [ ! -s patches_qemu ] || git apply $(cat patches_qemu )
+ENV VIRTIOFS_PATCHES_DIR=/root/kata_qemu/patches/${QEMU_VIRTIOFS_TAG}/
+RUN if [ -d ${VIRTIOFS_PATCHES_DIR} ]; then \
+	echo "Patches to apply for virtiofs fixes:"; \
+	for patch in $(find "${VIRTIOFS_PATCHES_DIR}" -name '*.patch' -type f |sort -t- -k1,1n); do \
+		git apply $patch; \
+	done;fi
```
Comment (jodh-intel) [inline static-build/qemu-virtiofs/Dockerfile:61]: Please change this block to:

- Add the `sort` functionality to `apply_patches.sh`.
- Call `apply_patches.sh $VIRTIOFS_PATCHES_DIR`.

**Review by jodh-intel — CHANGES_REQUESTED** [review]

**Comment by wainersm** [comment]
Hi @jodh-intel ! Since this is a back-port of already merged github.com/kata-containers/kata-containers/pull/787 , I think I should implement your suggestions on the original tree first so that the code do not diverge and future back-ports can be made easily. Any objection?

**Comment by jodh-intel** [comment]
@wainersm - agreed! I've raised https://github.com/kata-containers/kata-containers/issues/1014 to handle the improvements.

**Review by jodh-intel — APPROVED** [review]

**Comment by wainersm** [comment]
@jodh-intel @fidencio am I allowed to trigger the obs-packaging-ci job? If yes, how?

**Comment by wainersm** [comment]
@jodh-intel I decided to include the backport of https://github.com/kata-containers/kata-containers/pull/1025 in this same PR, so we have the change in a single place.

**Comment by wainersm** [comment]
The jenkins-ci-ubuntu-18-04 failed because of a checking in the kernel files:

```
16:37:38 Check Changes in kernel
16:37:38 kernel/build-kernel.sh
16:37:38 scripts/apply_patches.sh
16:37:38 static-build/qemu-virtiofs/Dockerfile
16:37:38 static-build/qemu/Dockerfile
16:37:38 Kernel directory has changes check kernel/kata_config_version changed
16:37:38 ERROR: Please bump version in kernel/kata_config_version
```
Should any updated file (including build scripts and README)  in the kernel directory drive to `kata_config_version` increment?

**Comment by jodh-intel** [comment]
> Should any updated file (including build scripts and README) in the kernel directory drive to `kata_config_version` increment?

@wainersm - yes: https://github.com/kata-containers/packaging/blob/master/.ci/test.sh#L39..L56 :)

**Review by jcvenegas — APPROVED** [review]

**Comment by jcvenegas** [comment]
@wainersm could you backport to stable-1.11 and stable-1.12 as well ?

**Comment by wainersm** [comment]
> @wainersm could you backport to stable-1.11 and stable-1.12 as well ?

yes @jcvenegas , I will backport to those branches as well.

**Pull request merged**